src/workerd/api/global-scope.c++

```diff
@@ -554,6 +556,47 @@ void ServiceWorkerGlobalScope::sendHibernatableWebSocketError(
   }
 }
 
+kj::Promise<WorkerInterface::CustomEvent::Result>
+ServiceWorkerGlobalScope::actorDestroy(Worker::Lock& lock,
+                                       kj::Maybe<ExportedHandler&> exportedHandler) {
+
+  auto& context = IoContext::current();
+  auto& handler = KJ_REQUIRE_NONNULL(exportedHandler);
+
+  if (handler.destroy == kj::none) {
+    lock.logWarningOnce("Calling actor destroy without any destroy() hander");
+    return WorkerInterface::CustomEvent::Result{.outcome = EventOutcome::SCRIPT_NOT_FOUND};
+  }
+
+  auto& destroy = KJ_ASSERT_NONNULL(handler.destroy);
+
+  auto fn = context.blockConcurrencyWhile(
+      lock,
+      [&context,
+       &destroy](jsg::Lock& js) mutable -> jsg::Promise<WorkerInterface::CustomEvent::Result> {
```

Comment:
If the destroy handler was defined to return a `jsg::Promise<void>` instead of a `kj::Promise<void>` the additional wrapping with `context.awaitIo(...)` below would be unnecessary.